x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/planning/AssignmentPlanner.java

```diff
@@ -63,11 +103,56 @@ public AssignmentPlan computePlan() {
         } else {
             bestPlan = planKeepingOneAllocationOnPreviousAssignments;
         }
-        logger.debug(() -> "Best plan =\n" + bestPlan.prettyPrint());
-        logger.debug(() -> prettyPrintOverallStats(bestPlan));
         return bestPlan;
     }
 
+    private AssignmentPlan solveAllocatingAtLeastOnceModelsThatWerePreviouslyAllocated() {
+        logger.debug(() -> "Attempting to solve assigning at least one allocations to previously assigned models");
+        List<Model> previouslyAssignedModelsOnly = models.stream()
```

Comment:
This may be a dumb question, but just to confirm. 

In the scenario where we already have allocated models and we re-calculate, `previouslyAssignedModelsOnly` also includes models that are CURRENTLY assigned. 

Basically the same as `solveSatisfyingPreviousAssignments()`, but also adding the restriction of at least one assignment for all models that have ever been allocated.

Reply:
Correct. Names are tricky. I'll have another think to see if I can come up with clearer terms.